src/events/events.ts

```diff
@@ -394,8 +395,14 @@ export const onPresenceUpdate = async (client: Client, before: Presence, after:
         }
     }
 
-    // Skip if no stream was found
-    if (!streams.length) {
+    // If a stream has ended, add a cooldown and skip
+    if (oldStreams.length > streams.length) {
+        startCooldown(uid);
+        return;
+    }
+
+    // Skip if the member is on cooldown or no stream was found
+    if (onCooldown(uid) || !streams.length) {
         return;
     }
```

Comment:
This is where cooldowns are applied and checked. The checks happen after the application, so cooldowns can be renewed.